### PR TITLE
Add demo for Excel automation in .NET Core 3.0

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
## Summary

Add sample for Excel automation in .NET Core 3.0 Preview 1.

cc @jeffschwMSFT @vitek-karas @richlander

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
I will add a new section in https://docs.microsoft.com/en-us/dotnet/samples-and-tutorials/ when this gets checked in.

@vitek-karas I created the `extensions` directory as a way to store projects that demonstrate how to use/design extensions using .NET. I would imagine that this would be a great location for examples on ALCs as well.

File: core/extensions/ExcelDemo/ExcelDemo.csproj [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+  </PropertyGroup>
+  
+  <!--
+    The 'COMReference' tags below were copied from a .NET Framework project that were inserted by Visual Studio.
```
Comment (richlander) [inline core/extensions/ExcelDemo/ExcelDemo.csproj]:
```suggestion
    The following 'COMReference' properties were copied from a .NET Framework project. They were added by using the Visual Studio COM References window. See: https://docs.microsoft.com/en-us/visualstudio/ide/managing-references-in-a-project?view=vs-2017
```

File: core/extensions/ExcelDemo/ReadMe.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Excel Demo
+================
+
+This is the basic Excel demo sample in .NET Core. It is designed to work with .NET Core 3.0 and beyond. The code example is a modification of the existing Office example of how to use [C# for Excel automation](https://support.microsoft.com/en-us/help/302084/how-to-automate-microsoft-excel-from-microsoft-visual-c-net).
```
Comment (richlander) [inline core/extensions/ExcelDemo/ReadMe.md]:
```suggestion
This is a basic Excel demo sample in .NET Core. It is designed to work with .NET Core 3.0. It is based on the [How to automate Microsoft Excel from Microsoft Visual C#.NET](https://support.microsoft.com/en-us/help/302084/how-to-automate-microsoft-excel-from-microsoft-visual-c-net) sample for .NET Framework.
```

File: core/extensions/ExcelDemo/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+Key Features
+------------
+
+Demonstrates how to consume Office PIAs in the .NET Core 3.0 Preview 1 timeframe.
```
Comment (richlander) [inline core/extensions/ExcelDemo/ReadMe.md]:
```suggestion
Demonstrates how to consume Office Primary Interop Assemblies (PIA) with .NET Core 3.0 Preview 1.
```

File: core/extensions/ExcelDemo/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+- Embedding of Interop types (i.e. [No-PIA](https://docs.microsoft.com/en-us/dotnet/framework/interop/type-equivalence-and-embedded-interop-types)).
+- Support for the [`IDispatch`](https://docs.microsoft.com/en-us/windows/desktop/winauto/idispatch-interface) interface.
+
+**Note** Adding COM references to .NET Core projects from Visual Studio is not currently supported. The workaround is to create a .NET Framework project, add the COM references, and then copy the relevant `COMReference` elements in the project. See `ExcelDemo.csproj` for further details.
+
+Build and Run
+-------------
+
+To build and run the sample, the project must be loaded in Visual Studio 2017. Developer support in `dotnet` is not complete for the Preview 1 release, but the scenario is demoable from within Visual Studio 2017.
```
Comment (richlander) [inline core/extensions/ExcelDemo/ReadMe.md]: What do you mean by "developer support"?

File: core/extensions/ExcelDemo/ReadMe.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+------------
+
+Demonstrates how to consume Office PIAs in the .NET Core 3.0 Preview 1 timeframe.
+
+- Embedding of Interop types (i.e. [No-PIA](https://docs.microsoft.com/en-us/dotnet/framework/interop/type-equivalence-and-embedded-interop-types)).
+- Support for the [`IDispatch`](https://docs.microsoft.com/en-us/windows/desktop/winauto/idispatch-interface) interface.
+
+**Note** Adding COM references to .NET Core projects from Visual Studio is not currently supported. The workaround is to create a .NET Framework project, add the COM references, and then copy the relevant `COMReference` elements in the project. See `ExcelDemo.csproj` for further details.
+
+Build and Run
```
Comment (richlander) [inline core/extensions/ExcelDemo/ReadMe.md]: Want to add command line instructions, too?

**Comment by AaronRobinsonMSFT** [comment]
@rpetrusha and @richlander Please take a look at the new changes. Thanks.

**Comment by mairaw** [comment]
@richlander should have permissions. Let me know if anyone else also needs it. We usually squash and merge our PRs.

**Comment by AaronRobinsonMSFT** [comment]
I am ready for a merge whenever. Thanks.

**Pull request merged**